core/server/proxy/src/main/java/alluxio/proxy/s3/signature/utils/AwsAuthV4HeaderParserUtils.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.proxy.s3.signature.utils;
+
+import alluxio.proxy.s3.S3ErrorCode;
+import alluxio.proxy.s3.S3Exception;
+import alluxio.proxy.s3.signature.AwsCredential;
+import alluxio.proxy.s3.signature.SignatureInfo;
+
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.StringTokenizer;
+
+/**
+ * Util to parse v4 auth information from header.
+ */
+public final class AwsAuthV4HeaderParserUtils {
+  private static final Logger LOG = LoggerFactory.getLogger(AwsAuthV4HeaderParserUtils.class);
+
+  private static final String CREDENTIAL = "Credential=";
+  private static final String SIGNEDHEADERS = "SignedHeaders=";
+  private static final String SIGNATURE = "Signature=";
+  private static final String AWS4_SIGNING_ALGORITHM = "AWS4-HMAC-SHA256";
+
+  /**
+   * Validate Signed headers.
+   *
+   * @param signedHeadersStr signed header
+   * @return a string which parsed singed headers
+   */
+  private static String parseSignedHeaders(String authHeader, String signedHeadersStr)
+          throws S3Exception {
+    if (StringUtils.isNotEmpty(signedHeadersStr)
+                && signedHeadersStr.startsWith(SIGNEDHEADERS)) {
+      String parsedSignedHeaders = signedHeadersStr.substring(SIGNEDHEADERS.length());
+      StringTokenizer tokenizer = new StringTokenizer(parsedSignedHeaders, ";");
+      if (tokenizer.hasMoreTokens()) {
+        return parsedSignedHeaders;
+      } else {
+        LOG.error("No signed headers found. AuthHeader:{}", authHeader);
+        throw new S3Exception(authHeader, S3ErrorCode.AUTHORIZATION_HEADER_MALFORMED);
+      }
+    } else {
+      LOG.error("No signed headers found. AuthHeader:{}", authHeader);
+      throw new S3Exception(authHeader, S3ErrorCode.AUTHORIZATION_HEADER_MALFORMED);
+    }
+  }
+
+  /**
+   * Function to parse signature.
+   * @param authHeader authorization header string
+   * @param dateHeader date string
+   * @return SignatureInfo instance
+   * @throws S3Exception
+   */
+  public static SignatureInfo parseSignature(String authHeader, String dateHeader)
```

Comment:
Can you provide an example of the format for a Signature in the comments here?

Reply:
finish.